test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -887,3 +887,15 @@ TEST_VM(os, is_first_C_frame) {
   EXPECT_FALSE(os::is_first_C_frame(&cur_frame));
 #endif // _WIN32
 }
+
+TEST_VM(os, open_O_CLOEXEC) {
+#if !defined(_WIN32)
+  int err = system("touch test.txt"); // create the "test.txt" file
```

Comment:
Where will the new test file get created and what will clean it up?
If it's created in the JTREG scratch dir, then I believe it is cleaned
up automatically.

Reply:
On Mach5 it would be created here:

/System/Volumes/Data/mesos/work_dir/slaves/0c72054a-24ab-4dbb-944f-97f9341a1b96-S122246/frameworks/1735e8a2-a1db-478c-8104-60c8b0af87dd-0196/executors/be4f78aa-5f35-474c-9b8b-ef5ea893c7d4/runs/9cea06d4-ecd1-401b-9487-0449f9297b1b/testoutput/test-support/jtreg_open_test_hotspot_jtreg_tier1_common/scratch/1

which is inside a "scratch" folder. When ran locally it goes here:

jdk/build/macosx-x86_64-server-release/test-support/gtest_all_server

Thanks.

Reply:
I'm concerned this may be fragile. If `system` does a fork/exec then we may have OOM issues. Is `touch` guaranteed to be available and in the path or else a built-in?

Can we not create the file via the appropriate API, then close it and then check os::open?

Reply:
I agree, this is fragile. Also, I think you don't need it. Just let the open that you test also create the file if it does not exist. E.g. `os::open("test", O_RDWR | O_CREAT | O_TRUNC, 0666);`. The "create file" and "open existing file" code paths are identical, so testing one should cover the other too.

Reply:
I like that, let me try it...